packages/react-sdk/src/Provider.tsx

```diff
@@ -14,17 +14,24 @@
  * limitations under the License.
  */
 import * as React from 'react'
+import * as optimizely from '@optimizely/optimizely-sdk'
 
 import { OptimizelyContextProvider } from './Context'
-import { OptimizelySDKWrapper, OptimizelyClientWrapper } from '@optimizely/js-web-sdk'
-import { UserAttributes, createUserWrapper, UserWrappedOptimizelySDK } from './createUserWrapper';
+import { ReactSDKClient } from './client'
+import { areUsersEqual } from './utils';
+
+type UserInfo = {
+  id: string
+  attributes?: optimizely.UserAttributes
+}
 
 interface OptimizelyProviderProps {
-  optimizely: OptimizelyClientWrapper
-  userId: string
+  optimizely: ReactSDKClient
   timeout?: number
-  userAttributes?: UserAttributes
   isServerSide?: boolean
+  user?: Promise<UserInfo> | UserInfo
```

Comment:
This is a weird pattern to use. Is this so that the user can provide some sort of resolver to us? I am not sure this would be a good thing to expose as a public API since I am just wary of using such distinct types together. We could always just require the developer to resolve the user before passing it in. Curious if this was an ask from a beta user

Reply:
I think this is valuable. The SDK consumer may want to fetch user info and wait until that fetch is complete before making decisions and rendering content based on those decisions.

Reply:
I think it's fair to ask how valuable this is, and to want a broader conversation about requirements for the whole SDK. To me this seems aligned with the current design of the SDK. We already handle rendering for different states of datafile availability, and this is extending the same functionality to user info.

Reply:
👍